src/recon.rs

```diff
@@ -2674,6 +2674,7 @@ pub(crate) unsafe fn rav1d_recon_b_intra<BD: BitDepth>(
                         );
 
                         if debug_block_info!(f, t.b) && DEBUG_B_PIXELS {
+                            let edge = edge_array.as_ptr().add(edge_offset);
                             hex_dump::<BD>(
                                 edge.offset(-(t_dim.h as isize * 4)),
```

Comment:
Can you combine these into `edge[edge_offset - t_dim.h as usize * 4..].as_ptr()`?  Same for the `edge` and `edge.add(1)` below?